src/telliot_feeds/reporters/tips/listener/tip_listener_filter.py

```diff
@@ -1,98 +1,109 @@
 import ast
-import sys
 from typing import Any
 from typing import Optional
 
 from clamfig.base import Registry
 from eth_abi import decode_single
-from eth_utils.conversions import to_bytes
 from web3 import Web3 as w3
 
 from telliot_feeds.feeds import CATALOG_FEEDS
 from telliot_feeds.feeds import DataFeed
 from telliot_feeds.feeds import DATAFEED_BUILDER_MAPPING
 from telliot_feeds.queries.query import OracleQuery
-from telliot_feeds.reporters.tips import CATALOG_QUERY_IDS
+from telliot_feeds.queries.query_catalog import query_catalog
 
 
-class TipListenerFilter:
-    """Check if query data supported"""
+def decode_typ_name(qdata: bytes) -> str:
```

Comment:
Might want to move these helpers to a more central place, since they could be used in a lot more places than the tip listener code.